src/typechecker.erl

```diff
@@ -317,6 +321,17 @@ compat_record_fields(_, _, _, _) ->
 ret(A) ->
     {A, constraints:empty()}.
 
+safe_any_type(_Ty, [], A, _TEnv) ->
```

Comment:
Please document this. I can't see what it does from just the name. What's safe?

Reply:
I figured out it's wrong anyway now. The point was to allow skipping keys that are not present, the side-effect right now is that it also allows a key to be there with the wrong type. I will make one for `assoc` and one for `exact` most likely and it'll make more sense.